replays/replays.lib.php

```diff
@@ -231,11 +231,12 @@ function upload(&$reqData) {
 			return 'not found';
 		}
 
+		$smogtours = preg_match('/^[a-z0-9]+-smogtours-[0-9]+$/', $id);
```

Comment:
Replay IDs are in the form of `<serverid>-<format>-<num>` (as evidenced by https://replay.pokemonshowdown.com/smogtours-ou-39893), so this seems to be a faulty regex.
```suggestion
		$smogtours = preg_match('/^smogtours-[a-z0-9]+-[0-9]+$/', $id);
```

Reply:
Thanks, I misread m[1]